src/frontends/pytorch/src/transforms/aten_stack_list_construct_replacer.cpp

```diff
@@ -38,22 +41,28 @@ AtenStackListConstructReplacer::AtenStackListConstructReplacer() {
         auto axis_node = pattern_map.at(axis).get_node_shared_ptr();
         auto axis_const = std::dynamic_pointer_cast<v0::Constant>(axis_node);
         auto axis = axis_const->cast_vector<int64_t>();
+        OPENVINO_ASSERT(axis.size() == 1);
         // Check if ListConstruct is an input
         if (auto list_construct_node = cast_fw_node(input_node, "prim::ListConstruct")) {
             const auto& list_inputs = list_construct_node->input_values();
-            OutputVector node_vector;
-            auto zero = v0::Constant::create(element::i32, Shape{}, {0});
-            // Iterate over values in ListConstruct
-            for (const auto& list_input : list_inputs) {
-                auto node = concat_list_construct(list_input);
-                auto unsqueezed_node = std::make_shared<v0::Unsqueeze>(node, axis_const);
-                node_vector.push_back(unsqueezed_node);
+            std::shared_ptr<Node> node;
+            if (auto compression = u4_compression_stack(nullptr, std::deque<Output<Node>>(list_inputs.begin(), list_inputs.end()), axis[0])) {
```

Comment:
`u4_compressoin_stack` is used in this place only, no need to use `deque` anymore.